src/controllers/api/monsterController.js

```diff
@@ -4,14 +4,51 @@ const utility = require('./utility');
 const { redisClient } = require('../../util');
 const getAsync = promisify(redisClient.get).bind(redisClient);
 
-exports.index = async (req, res, next) => {
-  const search_queries = {};
-  if (req.query.name !== undefined) {
-    search_queries.name = { $regex: new RegExp(utility.escapeRegExp(req.query.name), 'i') };
+/**
+ * Builds a MongoDB query based on provided URL query params in the request.
+ * Used to filter entries to those matching particular values
+ * @param queryParams URL query params in the API request
+ */
+function buildQuery(queryParams) {
+  const query = {};
+
+  if (queryParams.name) {
+    query.name = { $regex: new RegExp(utility.escapeRegExp(queryParams.name), 'i') };
+  }
+  if (queryParams.challenge_rating) {
+    query.challenge_rating = { $in: queryParams.challenge_rating.split(',') };
+  }
+  if (queryParams.alignment) {
+    query.alignment = { $regex: utility.csvToIRegExp(queryParams.alignment) };
+  }
```

Comment:
The alternative approach to this I've used has been:
```js
  if (req.query.alignment!== undefined) {
    const alignmentRegex = req.query.alignment.split(',').map(c => new RegExp(c, 'i'));
    search_queries.alignment = { $in: alignmentRegex };
  }
```
[Source](https://github.com/bagelbits/5e-srd-api/blob/master/src/controllers/api/spellController.js#L17-L20)

I'm not saying it's better, but some amount of consistency across them would be good. I'm open to either or a combination of both. The other approach still lets you use the `$in` parameter.

Reply:
I'm curious what the performance difference between one really big `$regex` vs `$in` filled with an array of Regex.

Reply:
You'll probably still want to make use of `utility.escapeRegExp`. Though most likely within `utility.csvtoIRegExp`

Reply:
Or even before you feed it to `csvToIRegExp`.